addons/website_slides/models/slide_channel.py

```diff
@@ -47,6 +48,38 @@ class ChannelUsersRelation(models.Model):
         )
     ]
 
+    def _compute_next_slide_id(self):
+        query = """
+            SELECT DISTINCT ON (partner_id, channel_id)
+                SS.id AS slide_id,
+                SS.channel_id AS channel_id,
+                p.id AS partner_id
+            FROM
+                slide_slide SS
+            JOIN res_partner p
+                 ON p.id IN %s
+            WHERE
+                SS.channel_id IN %s
+                AND SS.is_published = TRUE
+                AND SS.active = TRUE
+                AND NOT EXISTS (
+                    SELECT 1
+                      FROM slide_slide_partner
+                     WHERE slide_id = SS.id
+                       AND partner_id = p.id
+                       AND completed = TRUE
+                )
+            ORDER BY partner_id, channel_id, SS.sequence
+        """
+        self.env.cr.execute(query, [tuple(self.mapped('partner_id').ids), tuple(self.mapped('channel_id').ids)])
```

Comment:
won't we take too much record like that ?

if you compute the value for
- slide.channel.partner(id=5) (partner_id=6, channel_id=7)
- slide.channel.partner(id=8) (partner_id=9, channel_id=10)

we will compute the value for all `slide.slide.channel` where
```
partner_id IN (6, 9) and channel_id IN (7, 10)
```

it means, 4 possibles values instead of the 2 needed, if you see what I mean

With something like this,

```python
query = """
    SELECT DISTINCT ON (partner_id, channel_id)
           SCP.partner_id AS partner_id,
           SCP.channel_id AS channel_id,
           SS.id AS slide_id
      FROM slide_channel_partner SCP
      JOIN slide_slide SS
           ON SS.channel_id = SCP.channel_id
           AND SS.is_published = TRUE
           AND SS.active = TRUE
           AND NOT EXISTS(
                SELECT 1
                  FROM slide_slide_partner
                 WHERE slide_id = SS.id
                   AND partner_id = SCP.partner_id
                   AND completed = TRUE
            )
     WHERE SCP.id IN %s
    ORDER BY partner_id, channel_id, SS.sequence
"""
self.env.cr.execute(query, [tuple(self.ids)])
```

 the condition will be
```
(partner_id == 6 AND channel_id == 7)
OR (partner_id == 9 AND channel_id == 10)
```

which can give 2 record maximum instead of 4